include/xtensor/xfunction.hpp

```diff
@@ -1059,6 +1060,21 @@ namespace xt
         auto step_leading_lambda = [](auto&& st) { st.step_leading(); };
         for_each(step_leading_lambda, m_st);
     }
+
+    namespace detail
+    {
+        template<class F, class... CT>
+        struct has_fixed_size<xfunction<F, CT...>, std::enable_if_t<xt::detail::is_fixed<typename xt::xfunction<F, CT...>::shape_type>::value>>
```

Comment:
I think this introduces confusion, `has_fixed_size` and `get_fixed_size` should be dedicated to shape containers. An expression having a fixed size means the size of the underlying buffer is known at compile time, i.e. both the size and the *values* of the shape contianer are known at compile time. These metafunctions should propably be renamed in `has_fixed_dims` and `get_fixed_dims`

Reply:
Actually this is the intent, to use any type that has fixed size as a shape, xfunction can also represent a fixed sized shape, consider the following:
```cpp
auto shape = xt::adapt(std::array<size_t, 2>{}) / 2;
// shape is xfunction with fixed size 2
```

Reply:
Ok, so ignore my comment here.